src/catalog/src/local/manager.rs

```diff
@@ -332,24 +334,36 @@ impl CatalogManager for LocalCatalogManager {
                 schema_info: format!("{}.{}", catalog_name, schema_name),
             })?;
 
-        if schema.table_exist(&request.table_name)? {
-            return TableExistsSnafu {
-                table: format_full_table_name(catalog_name, schema_name, &request.table_name),
+        {
+            let _lock = self.register_lock.lock().await;
+            if let Some(existing) = schema.table(&request.table_name)? {
+                let exiting_table_id = existing.table_info().ident.table_id;
+                if exiting_table_id != request.table_id {
+                    return TableExistsSnafu {
+                        table: format_full_table_name(
+                            catalog_name,
+                            schema_name,
+                            &request.table_name,
+                        ),
+                    }
+                    .fail();
+                }
+                // Try to register table with same table id, just ignore.
+                Ok(0)
```

Comment:
Any particular thought we have to check whether table_id is the same? I think checking table_name is enough.

Reply:
Duplicated register for same table name with the same table id is ok and catalog manager just ignores it.
But registering tables with same table name but different table id is considered abnormal, which indicates that table engine has created same table name with different table id, which should be avoided by table engine.